packages/grid/_modules_/grid/models/props/DataGridProps.ts

```diff
@@ -244,20 +518,6 @@ interface GridComponentOtherProps extends CommonProps {
    * @param {GridCallbackDetails} details Additional details for this callback.
    */
   onColumnOrderChange?: GridEventListener<GridEvents.columnOrderChange>;
```

Comment:
Should this also be moved to `DataGridProProps.ts`? Users can only reorder columns in DataGridPro.

Reply:
Yes, I missed that one !

Reply:
The event is fired by `setColumnIndex` which is in `useGridColumns`
So technically, users can create a toolbar and use `apiRef.current.setColumnIndex` in it.
But we said we wanted to move it to `useGridColumnReorder`.
If we confirm that we'll do it, I can move `onColumnOrderChange` to the pro interface.

Reply:
Yes, they can but if they call in a toolbar while using DataGrid it won't work correctly because only DataGridPro listens for this event.

https://github.com/mui-org/material-ui-x/blob/6208886dbb143f7365f8387ef9d91a6261a3a9d3/packages/grid/x-data-grid-pro/src/DataGridProVirtualScroller.tsx#L159

`apiRef.current.setColumnIndex` could be understood as part of the column reorder API, so it wouldn't be available in DataGrid.

Reply:
Indeed
In that case I propose to move the code for `setColumnIndex` to `useGridColumnReorder` to clarify this buggy scenario and clean the typing

Reply:
It can be done later.